lib/theme.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * Returns a full config or one of the sections containing the
+ * setup for the theme found in a file, or a void array if none found.
+ *
+ * @param string $config_path Path to the config file.
+ * @param string $section_name Optional name of the section to pick.
+ * @return array Full theme config or its subset.
+ */
+function gutenberg_experimental_get_theme_config( $config_path, $section_name = null ) {
+	$empty_config = array();
+	if ( ! file_exists( $config_path ) ) {
+		return $empty_config;
+	}
+	$theme_config = json_decode(
+		file_get_contents( $config_path ),
+		true
+	);
+	if ( ! is_array( $theme_config ) ) {
+		return $empty_config;
+	}
+	if ( ! $section_name ) {
+		return $theme_config;
+	}
+	if ( ! isset( $theme_config[ $section_name ] ) || ! is_array( $theme_config[ $section_name ] ) ) {
+		return $empty_config;
+	}
+	return $theme_config[ $section_name ];
+}
+
+
+/**
+ * Extends block editor settings with the features loaded from the theme config.
+ *
+ * @param array $settings Default editor settings.
+ *
+ * @return array Filtered editor settings.
+ */
+function gutenberg_extend_settings_features( $settings ) {
+	$theme_features = gutenberg_experimental_get_theme_config(
```

Comment:
Can we find a way to pass `gutenberg_experimental_get_theme_config` all the keys that we want to retrieve so that we only execute this function once? Reading from the file system and `json_decode` can be expensive operations.

Reply:
How about we don't use JSON at all for the default theme config? 😄 

I'm serious here. If you combine this feedback with https://github.com/WordPress/gutenberg/pull/22291/files#r423893362 – the issue with the JSON file in the root folder, it raises the question of why we opted for JSON for WordPress core in the first place. Do we plan to share this file with the client?

Reply:
To be honest, I don't remember. Perhaps in some early explorations we pass it directly to the client? It seems fine to me that we go with PHP for this. cc @jorgefilipecosta @ItsJonQ in case they have better memories.